File: HACKERRANK/java/easy/plus_minus.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<Integer> arr) {
        float value_counts[] = new float[3];
        value_counts[0] = 0; // positive
        value_counts[1] = 0; // negative
        value_counts[2] = 0; // zero

        for (int i = 0; i < arr.size(); i++) {
            if (arr.get(i) > 0) {
                value_counts[0] += 1;
            } else if (arr.get(i) < 0) {
                value_counts[1] += 1;
            } else {
                value_counts[2] += 1;
            }
        }

        for (int i = 0; i < value_counts.length; i++) {
            System.out.println(value_counts[i] / arr.size());
        }
    }

}

public class plus_minus {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                .map(Integer::parseInt)
                .collect(toList());

        Result.plusMinus(arr);

        bufferedReader.close();
    }
}

File: HACKERRANK/python/easy/plus_minus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    
    n = len(arr)
    proportion=[
        0, # positive
        0, # negative
        0  # zero
    ]
    
    # logic
    for i in range(n):
        if arr[i] > 0:
            proportion[0] += 1
        elif arr[i] < 0:
            proportion[1] += 1
        elif arr[i] == 0:
            proportion[2] += 1
    
    # print(proportion[0])
    # print(proportion[1])
    # print(proportion[2])
    proportion[0] /= n
    proportion[1] /= n
    proportion[2] /= n
    
    for i in proportion:
        print(i)
    
    
if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)
